src/main/java/software/amazon/cloudformation/LambdaWrapper.java

```diff
@@ -464,13 +471,27 @@ protected void validateModel(final JSONObject modelObject) throws ValidationExce
     private void publishExceptionMetric(final Action action, final Throwable ex, final HandlerErrorCode handlerErrorCode) {
         if (this.metricsPublisherProxy != null) {
             this.metricsPublisherProxy.publishExceptionMetric(Instant.now(), action, ex, handlerErrorCode);
+            // always puts failed events
+            publishExceptionCodeAndCountMetric(action, handlerErrorCode, true);
```

Comment:
We have 3 places to publish the metrics for the error code.  I think we should merge them into one, just to have it after `wrapInvocationAndHandleErrors`.